src/uipath/agent/models/agent.py

```diff
@@ -645,83 +459,191 @@ class AgentBuiltInValidatorGuardrail(BuiltInValidatorGuardrail):
+class AgentUnknownGuardrail(BaseCfg):
+    """Fallback wrapper for unknown guardrail kinds."""
+
+    guardrail_type: Literal["unknown"] = Field(alias="$guardrailType")
+    # store the original payload under 'raw' for round-trip/debug
+    raw: Dict[str, Any]
+
+
 AgentGuardrail = Annotated[
-    Union[AgentCustomGuardrail, AgentBuiltInValidatorGuardrail],
+    Union[
+        # known kinds from SDK
+        AgentCustomGuardrail,
+        AgentBuiltInValidatorGuardrail,
+        # unknown kind fallback
+        AgentUnknownGuardrail,
+    ],
     Field(discriminator="guardrail_type"),
 ]
 
 
-class BaseAgentDefinition(BaseModel):
-    """Agent definition model."""
+class AgentMetadata(BaseCfg):
+    """Agent metadata model."""
 
-    input_schema: Dict[str, Any] = Field(
-        ..., alias="inputSchema", description="JSON schema for input arguments"
-    )
-    output_schema: Dict[str, Any] = Field(
-        ..., alias="outputSchema", description="JSON schema for output arguments"
-    )
-    guardrails: Optional[List[AgentGuardrail]] = Field(
-        None, description="List of agent guardrails"
-    )
-
-    model_config = ConfigDict(
-        validate_by_name=True, validate_by_alias=True, extra="allow"
-    )
+    is_conversational: bool = Field(alias="isConversational")
+    storage_version: str = Field(alias="storageVersion")
 
 
-class AgentType(str, Enum):
-    """Agent type."""
+class AgentMessage(BaseCfg):
+    """Agent message model."""
 
-    LOW_CODE = "lowCode"
-    CODED = "coded"
+    role: Literal[AgentMessageRole.SYSTEM, AgentMessageRole.USER]
+    content: str
 
+    @field_validator("role", mode="before")
+    @classmethod
+    def normalize_role(cls, v: Any) -> Any:
+        """Normalize role to lowercase enum/string."""
+        return v.lower() if isinstance(v, str) else v
 
-class LowCodeAgentDefinition(BaseAgentDefinition):
-    """Low code agent definition."""
 
-    type: Literal[AgentType.LOW_CODE] = AgentType.LOW_CODE
+class AgentSettings(BaseCfg):
+    """Agent settings model."""
 
-    id: str = Field(..., description="Agent id or project name")
-    name: str = Field(..., description="Agent name or project name")
-    metadata: Optional[AgentMetadata] = Field(None, description="Agent metadata")
-    messages: List[AgentMessage] = Field(
-        ..., description="List of system and user messages"
-    )
+    engine: str
+    model: str
+    max_tokens: int = Field(..., alias="maxTokens")
+    temperature: float
 
-    version: str = Field("1.0.0", description="Agent version")
-    resources: List[AgentResourceConfig] = Field(
-        ..., description="List of tools, context, mcp and escalation resources"
-    )
-    features: List[Any] = Field(default_factory=list, description="Agent feature list")
-    settings: AgentSettings = Field(..., description="Agent settings")
 
-    model_config = ConfigDict(
-        validate_by_name=True, validate_by_alias=True, extra="allow"
-    )
+class AgentDefinition(BaseModel):
```

Comment:
I was thinking we need to remove `messages` , `settings`  etc from there since those are applicable only in low-code scenarios. The base agent definition (could contain only the entrypoint, resources etc) and let lowcode/coded agents etc extend from that.

Also note that Resource is a superset of bindings and contains useful details such as I/O schema. We could deterministically convert from resource to bindings object.

Basically, IMO the AgentDefinition should have a 1:1 mapping with a Runtime and we could send the agent definition to the runtime object during runtime object construction. i.e; `agent = LowCodeAgentRuntime(agent_definition=low_code_agent_definition)`

As far as `LowCode` is concerned, maybe `ReAct` is a better name for this.